### ISSUE TITLE
Docs: broken '[source]' link to external (cfdm) methods

### ISSUE THREAD

**Issue by sadielbartholomew** [NCAS-CMS/cf-python#34]
In the cf-python v. 3.1.0 (& cfdm v. 1.7.11) documentation, clicking a '[source]' link for a method under the API Reference where the code from the method originates externally (in practice, in the cfdm library), will lead to a 404 error page. This is because it is attempting to access the code via the cf-python package's environment. To illustrate, using the examples comparing [this current reference page](https://ncas-cms.github.io/cf-python/method/cf.AuxiliaryCoordinate.has_bounds.html#cf.AuxiliaryCoordinate.has_bounds) and [its 3.0.6 equivalent](https://ncas-cms.github.io/cf-python/3.0.6/method/cf.AuxiliaryCoordinate.get_bounds.html?highlight=get_bounds#cf.AuxiliaryCoordinate.get_bounds):

* in the cf-python 3.0.6 docs, this text linked successfully to:
`https://github.com/NCAS-CMS/cfdm/blob/v1.7.11/cfdm/mixin/propertiesdatabounds.py#L870`
* now it links to a URL with some environment info included, which breaks the link, e.g: `https://github.com/NCAS-CMS/cf-python/blob/anaconda3/envs/cf-env/lib/python3.8/site-packages/cfdm/core/abstract/propertiesdatabounds.py#L387`

**Comment by davidhassell** [comment]
Could this be due to a deficiency in the filename inspection during the documentation build, coming from https://github.com/NCAS-CMS/cf-python/blob/master/docs/source/conf.py#L466 ?

**Comment by sadielbartholomew** [comment]
> Could this be due to a deficiency in the filename inspection during the documentation build, coming from https://github.com/NCAS-CMS/cf-python/blob/master/docs/source/conf.py#L466 ?

That seems very plausible, thanks for the tip. I'll look into it next week after the training course.

**Comment by sadielbartholomew** [comment]
I'll pick this up today to fix before the release.

**Comment by sadielbartholomew** [comment]
Fixed (in 00b2c7992d35c44479dc785947a63547e7931c14), though it won't show the fixed links until a re-build is committed, namely for the release.

FYI @davidhassell you were correct in your suspicion. The statement to process the ``cfdm`` files was being skipped because my system ``cfdm`` installation upon build was not my git clone but a pip install in my conda environment, hence the ``fn`` always had ``../../anaconda3/lib/python3.7/site-packages/cfdm/`` prepended to the method name, rather than ``<root path>/cfdm/cfdm/`` (the keyword twice from the repo hierarchy) as it would have been in the former case.

I checked & found testing for ``cfdm/`` instead was sufficient, particularly not picking up any files not related to ``cfdm`` from the more lenient search.
